sql/parser/normalize.go

```diff
@@ -119,19 +122,25 @@ func (expr *ComparisonExpr) normalize(v *normalizeVisitor) Expr {
 				//   /     \            /     \
 				//  a       1          2       1
 
+				rotating := true
 				switch left.Operator {
-				case Plus, Minus, Div:
+				case Plus:
+					left.Operator = Minus
+				case Minus:
+					left.Operator = Plus
+				case Div:
+					left.Operator = Mult
+				default:
+					rotating = false
+				}
+
+				if rotating {
+					// Clear the function caches since we're rotating.
+					expr.fn.fn = nil
+					left.fn.fn = nil
+
```

Comment:
Yeah, this is the second commit, and it is equivalent. I find this easier to read (and I wanted to similarly refactor the other case below, but couldn't figure it out.

I'll leave it out for now.